# Lab-7


# Hypothesis Part

* Name: Guo Xinfu
* Student ID: n01611988

## Import Libraries

In [16]:
from scipy import stats
import numpy as np

### Choose the best Test and apply.

## Exercise 1

suppose we want to know whether or not the mean weight of a certain species of some turtle is equal to 310 pounds.

To test this, we go out and collect a simple random sample of turtles with the following weights:

Weights: 300, 315, 320, 311, 314, 309, 300, 308, 305, 303, 305, 301, 303

In [22]:
from scipy.stats import ttest_1samp
    
weight = [300, 315, 320, 311, 314, 309, 300, 308, 305, 303, 305, 301, 303]
mean=np.mean(weight)
print("mean:", mean)
t_test, p_val = ttest_1samp(weight, 310)
print("t_test:",t_test)
print("p_val:",p_val)

if p_val < 0.05 :
    print("we can reject the null hypothesis")
else:
    print("we fail to reject the null hypothesis")

mean: 307.2307692307692
t_test: -1.5848116313861254
p_val: 0.1389944275158753
we fail to reject the null hypothesis


This means that there is not sufficient evidence to conclude that the mean weight of the turtle species is different from 310 pounds based on the provided sample data.

## Exercise 2


suppose we want to know whether or not the mean weight between two different species of turtles is equal.

To test this, we collect a simple random sample of turtles from each species with the following weights:

Sample 1: 300, 315, 320, 311, 314, 309, 300, 308, 305, 303, 305, 301, 303

Sample 2: 335, 329, 322, 321, 324, 319, 304, 308, 305, 311, 307, 300, 305

In [20]:
from scipy.stats import ttest_ind

sample1=np.array([300, 315, 320, 311, 314, 309, 300, 308, 305, 303, 305, 301, 303])
sample2=np.array([335, 329, 322, 321, 324, 319, 304, 308, 305, 311, 307, 300, 305])

mean1=np.mean(sample1)
mean2=np.mean(sample2)
sdt1=np.std(sample1)
sdt2=np.std(sample2)
print("mean of sample1:", mean1)
print("mean of sample2:", mean2)
print("standard deviation of sample1:", sdt1)
print("standard deviation of sample2:", sdt2)

t_test, p_val = ttest_ind(sample1, sample2)
print("t_test:", t_test)
print("p_val:", p_val)

if p_val < 0.05 :
    print("we can reject the null hypothesis")
else:
    print("we fail to reject the null hypothesis")

mean of sample1: 307.2307692307692
mean of sample2: 314.61538461538464
standard deviation of sample1: 6.053020176278769
standard deviation of sample2: 10.565098285162705
t_test: -2.10090292575557
p_val: 0.04633501389516513
we can reject the null hypothesis


This suggests that there is sufficient evidence to conclude that the mean weight of the turtle species represented by Sample 1 is significantly different from the mean weight represented by Sample 2.

## Exercise 3


In [ ]:
The following data shows the max jump height (in inches) before and after using the training program for each player:

Before: 22, 24, 20, 19, 19, 20, 22, 25, 24, 23, 22, 21

After: 23, 25, 20, 24, 18, 22, 23, 28, 24, 25, 24, 20

In [23]:
before = [22, 24, 20, 19, 19, 20, 22, 25, 24, 23, 22, 21]
after = [23, 25, 20, 24, 18, 22, 23, 28, 24, 25, 24, 20]

stats.ttest_rel(before,after)

TtestResult(statistic=-2.5289026942943655, pvalue=0.02802807458682508, df=11)

Statistic: The t-statistic is -2.5289026942943655. This value indicates how many standard errors the sample mean is from the null hypothesis mean. A negative value suggests that the sample mean is below the null hypothesis mean.

P-value: The p-value is 0.02802807458682508. This value represents the probability of observing a t-statistic as extreme as or more extreme than the one obtained from the sample, assuming that the null hypothesis (that the mean weight of the turtle species is equal to 310 pounds) is true. A low p-value (typically below the chosen significance level, often 0.05) suggests that there is enough evidence to reject the null hypothesis.

Degrees of Freedom (df): The degrees of freedom is 11. This value represents the number of independent pieces of information available to estimate a parameter. It's commonly used in the calculation of the t-statistic.